src/auth/ui/vue/show.ts

```diff
@@ -79,8 +78,25 @@ export class AuthWebview extends VueWebview {
         return getCredentialsErrors(data)
     }
 
-    async trySubmitCredentials(profileName: SectionName, data: StaticProfile) {
-        return tryAddCredentials(profileName, data, true)
+    /**
+     * @returns true if successfully added credentials
+     */
+    async trySubmitCredentials(profileName: SectionName, data: StaticProfile): Promise<boolean> {
+        try {
+            await tryAddCredentials(profileName, data, true)
+            return true
+        } catch (e) {
+            if (!(e instanceof Error)) {
+                return false
+            }
+            telemetry.auth_addConnection.emit({
+                source: this.getSource() || '',
+                credentialSourceId: 'sharedCredentials',
+                result: 'Failed',
+                reason: e.message,
+            })
+            return false
+        }
```

Comment:
Use `run` when instrumenting workflows:
```suggestion
        try {
            await telemetry.auth_addConnection.run(span => {
                span.record({
                    source: this.getSource() || '',
                    credentialSourceId: 'sharedCredentials' 
                })
            
                return tryAddCredentials(profileName, data, true)    
            })
        
            return true
        } catch {
            return false
        }
```

Reply:
Removed this emit, not needed anymore